packages/beacon-node/src/network/reqresp/utils/dataColumnResponseValidaiton.ts

```diff
@@ -0,0 +1,66 @@
+import {LogData} from "@lodestar/logger";
+import {ForkSeq, KZG_COMMITMENTS_GINDEX} from "@lodestar/params";
+import {RespStatus, ResponseError} from "@lodestar/reqresp";
+import {ColumnIndex, Slot} from "@lodestar/types";
+import {Logger, bytesToInt, toHex} from "@lodestar/utils";
+import {IBeaconChain} from "../../../chain/interface.js";
+import {IBeaconDb} from "../../../db/interface.js";
+
+export async function logDataColumnSidecarUnavailability(opts: {
+  chain: IBeaconChain;
+  db: IBeaconDb;
+  index: ColumnIndex;
+  slot: Slot;
+  blockRoot?: Uint8Array;
+  logData: Record<string, LogData>;
+}): Promise<void> {
+  const {chain, db, index, slot, blockRoot, logData} = opts;
+  let localLogData: LogData = {...logData, slot};
+
+  if (blockRoot) {
+    localLogData = {...localLogData, blockRoot: toHex(blockRoot)};
+  }
+
+  const blockBytes = blockRoot ? await db.block.getBinary(blockRoot) : await db.blockArchive.getBinary(slot);
+
+  if (!blockBytes) {
+    chain.logger.error("Expected unfinalized block not found", localLogData);
+    return;
+  }
+
+  const blobsCount =
+    chain.config.getForkSeq(slot) < ForkSeq.deneb
```

Comment:
in what scenario is this code executed pre-deneb? if we are unsure about the fork here (which we shouldn't be) then rather assert and throw here as this seems like a bug in our code